### PR TITLE
Add SQL upgrade queries for new unit price column

### PR DESCRIPTION
**Pull request by jolelievre**
| Questions         | Answers
| ------------------| -------------------------------------------------------
| Description?      | These upgrade queries are related to this PR https://github.com/PrestaShop/PrestaShop/pull/26762 which adds a new column in product tables<br>In 178 the product table used the `unit_price_ratio` to save the unit price, so the actual value was not saved only the ratio instead. In 8.0.0 we now have a `unit_price` column which persists the actual value specified in the BO and used in the FO. But we kept the `unit_price_ratio` for backward compatibility for now.<br><br>This upgrade PR does to things:<br>- it adds the new column `unit_price` in the database<br>- it computes its value based on existing `unit_price_ratio` and `price` column
| Type?             | improvement
| BC breaks?        | no
| Deprecations?     | no
| Fixed ticket?     | ~
| How to test?      | You should create (or edit) some products with defined prices AND unit prices. After the upgrade check that the `unit_price` column has been created in the `product` table and `product_shop` table, the `unit_price` values should be prefilled based on previous values. The formula to compute it is the following `unit_price = price / unit_price_ratio`, when you go back to the product edition the value for the unit price should be the same even if it is now based on a different column.
| Possible impacts? | ~

### CODE REVIEW

**Review by PierreRambaud — APPROVED** [review]

**Comment by kpodemski** [comment]
@jolelievre rebase needed 👍

**Review by kpodemski — CHANGES_REQUESTED** [review]
To be rebased after merge of the related PR from core

**Review by PierreRambaud — DISMISSED** [review]
PR has been rebased

**Review by Progi1984 — APPROVED** [review]

**Comment by marwachelly** [comment]
Hello,

This PR cannot be tested now. It is blocked by the autoupgrade issue #26503 since the PR related is on develop branch. It will be tested as soon as the issue will be fixed.

Thanks,

**Comment by khouloudbelguith** [comment]
Hi @jolelievre, @marwachelly,

As discussed with @atomiix while testing the blocked PR
There are some bugs should be fixed in this PR as commented here: https://github.com/PrestaShop/autoupgrade/pull/434#pullrequestreview-884427215

Thank you!

**Comment by marwachelly** [comment]
Hello @jolelievre ,

Could you please check the conflicted files?

Thanks,

**Review by Progi1984 — CHANGES_REQUESTED** [review]
@jolelievre Could you rebase your PR ?

**Review by Progi1984 — APPROVED** [review]

File: upgrade/sql/8.0.0.sql
Comment (atomiix) [inline upgrade/sql/8.0.0.sql:56]: Shouldn't we use the PrestaShop round type here?

Comment (jolelievre) [reply upgrade/sql/8.0.0.sql:56]: I'm not sure, the mysql column size will automatically truncate the values anyway. As far as I know the round parameter are used during computation to decide how to round the value for display but not in the column updates. At least for this new column I didn't integrate some rounding before DB update in order to keep as many decimals as possible.

**Review by atomiix — APPROVED** [review]

**Review by khouloudbelguith — COMMENTED** [review]
Hello @jolelievre,

I followed these steps:

1. Install PS1786 with English language & France country
2. Uninstall some incompatible modules
3. Install the upgrade module from this PR
4. Upgrade to 8.0.x
5. After the upgrade, I'm blocked, I cannot test the product V2 because it is hidden

https://user-images.githubusercontent.com/16067358/173553007-b5360545-00e2-4f44-a344-c7a5d2264f7a.mp4

We should wait for this PR: https://github.com/PrestaShop/autoupgrade/pull/489 to be merged

Thanks!

**Review by khouloudbelguith — COMMENTED** [review]
Hi @jolelievre,

The [PR](https://github.com/PrestaShop/autoupgrade/pull/489) is merged.
Could you please check the Conflicting files?

Thanks!

**Comment by jolelievre** [comment]
Thanks @khouloudbelguith 

the branch is rebased no content was modified only addressed the conflict because a PR that modified the same place was merged. It should be testable even if I'm waiting for a re-approval

**Review by Progi1984 — APPROVED** [review]

**Review by atomiix — APPROVED** [review]

**Pull request merged**
